fitbenchmarking/parsing/fitbm_problem.py

```diff
@@ -0,0 +1,150 @@
+# Copyright &copy; 2016 ISIS Rutherford Appleton Laboratory, NScD
+# Oak Ridge National Laboratory & European Spallation Source
+#
+# This file is part of Mantid.
+# Mantid is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# Mantid is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# File change history is stored at: <https://github.com/mantidproject/mantid>.
+# Code Documentation is available at: <http://doxygen.mantidproject.org>
+from __future__ import (absolute_import, division, print_function)
+
+from abc import ABCMeta, abstractmethod
+
+
+class BaseFittingProblem(object):
+    """
+    Definition of a base class implementation of the fitting test problem,
+    normally loaded from a problem definition file.
+
+    Types of data:
+        - strings: name, type, equation
+        - floats: start_x, end_x, ref_residual_sum_sq
+        - numpy arrays: data_x, data_y, data_e
+        - arrays: starting_values
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, fname):
+
+        # Initializes base class with file and problem type
+        self.fname = fname
```

Comment:
why is fname attribute without _ when others are not, not clear from just quickly looking at the code